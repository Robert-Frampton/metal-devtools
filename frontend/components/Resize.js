import Component, {Config} from 'metal-jsx';

class Resize extends Component {
	created() {
		this.doDrag = this.doDrag.bind(this);
		this.initDrag = this.initDrag.bind(this);
		this.stopDrag = this.stopDrag.bind(this);
	}

	attached() {
		this.element.addEventListener('mousedown', this.initDrag, false);
	}

	doDrag({clientX}) {
		this.props.onResize({clientX});
	}

	stopDrag() {
		document.body.removeEventListener('mousemove', this.doDrag, false);
		document.body.removeEventListener('mouseup', this.stopDrag, false);
	}

	initDrag() {
		document.body.addEventListener('mousemove', this.doDrag, false);
		document.body.addEventListener( 'mouseup', this.stopDrag, false);
	}

	render() {
		return (
			<div class="resize-container" />
		);
	}
}

Resize.PROPS = {
	onResize: Config.func().required()
};

export default Resize;
