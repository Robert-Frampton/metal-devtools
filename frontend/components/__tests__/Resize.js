jest.unmock('../Resize');

import Resize from '../Resize';

describe('Resize', () => {
	it('should render', () => {
		const component = new Resize({onResize: jest.fn()});

		expect(snap(component)).toMatchSnapshot();
	});
});
