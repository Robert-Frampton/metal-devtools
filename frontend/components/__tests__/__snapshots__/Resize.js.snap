exports[`Resize should render 1`] = `undefined`;
